srk/src/termination.ml

```diff
@@ -0,0 +1,429 @@
+    )
+    xpp_list; *)
+  (* let () = BatList.iter (function
+  | `LeqZero t -> Log.errorf "%a <= 0" (CoordinateSystem.pp_vector cs) t
+  | `LtZero t -> Log.errorf "%a < 0" (CoordinateSystem.pp_vector cs) t
+  | `EqZero t -> Log.errorf "%a = 0" (CoordinateSystem.pp_vector cs) t) header_eqs
+  in *)
+  (* let () = BatList.iter (function
+  | `LeqZero t -> Log.errorf "%a <= 0" (CoordinateSystem.pp_vector cs) t
+  | `LtZero t -> Log.errorf "%a < 0" (CoordinateSystem.pp_vector cs) t
+  | `EqZero t -> Log.errorf "%a = 0" (CoordinateSystem.pp_vector cs) t) body_eqs
+  in *)
+  let constraints = BatList.concat [header_eqs; body_eqs] in
+  let () = BatList.iter (function
+  | `LeqZero t -> logf "%a <= 0" (CoordinateSystem.pp_vector cs) t
+  | `LtZero t -> logf "%a < 0" (CoordinateSystem.pp_vector cs) t
+  | `EqZero t -> logf "%a = 0" (CoordinateSystem.pp_vector cs) t) constraints
+  in
+  let mat_a, n_rows = get_coeff_matrix cs xp_list constraints in
+  let mat_ap, n_rows2 = get_coeff_matrix cs xpp_list constraints in
+  let vec_b, n_rows3 = get_constants_vec cs constraints in
+  let mat_b = Mat.zero in
+  let mat_b = Mat.add_column 0 vec_b mat_b in
+    (* Format.fprintf Format.std_formatter "\nMatrix A:\n";
+    (Mat.pp Format.std_formatter mat_a);
+    Format.fprintf Format.std_formatter "\nMatrix A':\n";
+    (Mat.pp Format.std_formatter mat_ap);
+    Format.fprintf Format.std_formatter "\nMatrix b:\n";
+    (Mat.pp Format.std_formatter mat_b); *)
+  let n_cols = List.length xp_list in
+  let lambda1s = create_const_symbols srk "l" n_rows in
+  let lambda2s = create_const_symbols srk "r" n_rows in
+  let subtractions = build_subtracted_lambdas srk lambda1s lambda2s in
+  let l1gt0 = build_gt_zero srk lambda1s in
+  print_formula_to_stdout srk "\nConstraints lambda_1 >= 0:\n" l1gt0;
+  let l2gt0 = build_gt_zero srk lambda2s in
+  print_formula_to_stdout srk "\nConstraints lambda_2 >= 0:\n" l2gt0;
+  let eq1 = build_formula srk lambda1s mat_ap n_rows n_cols "eq_0" in
+  print_formula_to_stdout srk "\nConstraints eq1:\n" eq1;
+  
+  let eq2 = build_formula srk subtractions mat_a n_rows n_cols "eq_0" in
+  print_formula_to_stdout srk "\nConstraints eq2:\n" eq2;
+
+  let eq3 = build_formula srk lambda2s (Mat.add mat_a mat_ap) n_rows n_cols "eq_0" in
+  print_formula_to_stdout srk "\nConstraints eq3:\n" eq3;
+
+  let eq4 = build_formula srk lambda2s mat_b n_rows 1 "lt_0" in
+  print_formula_to_stdout srk "\nConstraints eq4:\n" eq4;
+
+  let all_eqs = mk_and srk [l1gt0; l2gt0; eq1; eq2; eq3; eq4] in
+  let rf_terms_list = build_lambda_mat_prod_terms srk lambda2s mat_ap n_rows n_cols in
+  (* List.iter (fun x -> Syntax.Term.pp srk Format.std_formatter x; Format.fprintf Format.std_formatter "\n" ) rf_terms_list; *)
+  let rf_delta0 = build_lambda_mat_prod_terms srk lambda1s mat_b n_rows 1 in
+  let rf_delta = build_lambda_mat_prod_terms srk lambda2s mat_b n_rows 1 in
+  (* let solver = SrkZ3.mk_solver srk in
```

Comment:
Generally speaking: strongly prefer not to commit code that's been commented out.